test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,528 @@
+    } else {
+        error!("Error: {field} field is not a string");
+        Err(serde::de::Error::custom(format!(
+            "Error: {field} field is not a string"
+        )))
+    }
+}
+
+fn parse_u32_field(value: &Value, field: &str) -> Result<Option<u32>, serde_json::Error> {
+    if let Some(entity_value) = value
+        .get("entity")
+        .and_then(|entity_value| entity_value.get(field))
+        .and_then(|field_value| field_value.as_str())
+    {
+        if let Ok(parsed_value) = entity_value.parse::<u32>() {
+            Ok(Some(parsed_value))
+        } else {
+            error!("Error: {field} is not a number");
+            Err(serde::de::Error::custom(format!(
+                "Error: {field} is not a number"
+            )))
+        }
+    } else {
+        error!("Error: {} field is not a string", field);
+        Err(serde::de::Error::custom(format!(
+            "Error: {field} field is not a string"
+        )))
+    }
+}
+
+fn parse_uentity(value: &Value) -> UEntity {
+    let name = match parse_string_field(value, "name") {
+        Ok(value) => value,
+        Err(_) => "None".to_owned(),
+    };
+
+    let id = match parse_u32_field(value, "id") {
+        Ok(value) => value,
+        Err(_) => None,
+    };
+
+    let version_major = match parse_u32_field(value, "version_major") {
+        Ok(value) => value,
+        Err(_) => None,
+    };
+
+    let version_minor = match parse_u32_field(value, "version_minor") {
+        Ok(value) => value,
+        Err(_) => None,
+    };
+
+    UEntity {
+        name,
+        id,
+        version_major,
+        version_minor,
+        special_fields: SpecialFields::default(),
+    }
+}
+
+fn parse_uauthority(value: &Value) -> Result<UAuthority, serde_json::Error> {
+    let name = if let Some(authority_value) = value
+        .get("authority")
+        .and_then(|authority_value| authority_value.get("name"))
+        .and_then(|authority_value| authority_value.as_str())
+    {
+        Some(authority_value.to_owned())
+    } else {
+        return Err(serde::de::Error::custom("Missing value authority name"));
+    };
+
+    let number = if let Some(authority_number_ip) = value
+        .get("authority")
+        .and_then(|authority_value| authority_value.get("number"))
+        .and_then(|number| number.get("ip"))
+    {
+        Some(up_rust::Number::Ip(
+            authority_number_ip.to_string().as_bytes().to_vec(),
+        ))
+    } else if let Some(authority_number_id) = value
+        .get("authority")
+        .and_then(|authority_value| authority_value.get("number"))
+        .and_then(|number| number.get("id"))
+    {
+        Some(up_rust::Number::Id(
+            authority_number_id.to_string().as_bytes().to_vec(),
+        ))
+    } else {
+        return Err(serde::de::Error::custom("Missing value number"));
+    };
+
+    Ok(UAuthority {
+        name,
+        number,
+        special_fields: SpecialFields::default(),
```

Comment:
Prefer using the Default trait impl here to set the special_fields as they are only used for book-keeping internal to the protobuf crate:
```suggestion
        Default::default(),
```

Reply:
Why was this one marked as resolved? Seems like to me this wasn't done?

Reply:
I resolved it because of a clippy error I was getting... now, I see that, with the two dots, I don't get the clippy error